File: python/tests/tools_test.py
"""Tests for tools package."""

import sys
sys.path.insert(0, "src")

from pedro_agentware.tools import Tool, Result, BaseTool, ToolRegistry


class AddTool(BaseTool):
    @property
    def name(self) -> str:
        return "add"

    @property
    def description(self) -> str:
        return "Add two numbers"

    def execute(self, args: dict) -> Result:
        a = args.get("a", 0)
        b = args.get("b", 0)
        return Result(success=True, data=a + b)


def test_tool_registry_register():
    registry = ToolRegistry()
    tool = AddTool()
    registry.register(tool)
    assert tool.name in registry.names()


def test_tool_registry_get():
    registry = ToolRegistry()
    tool = AddTool()
    registry.register(tool)
    found, ok = registry.get("add")
    assert ok
    assert found.name == "add"


def test_tool_registry_all():
    registry = ToolRegistry()
    tool = AddTool()
    registry.register(tool)
    all_tools = registry.all()
    assert len(all_tools) == 1


def test_result_to_dict():
    result = Result(success=True, data=42)
    d = result.to_dict()
    assert d["success"] is True
    assert d["data"] == 42

File: typescript/tests/tools.test.ts
import { ToolRegistry, BaseTool, Result } from "../src/tools/index.js";

class AddTool extends BaseTool {
  constructor() {
    super("add", "Add two numbers");
  }

  execute(args: Record<string, unknown>): Result {
    const a = (args.a as number) || 0;
    const b = (args.b as number) || 0;
    return new Result(true, a + b);
  }
}

describe("ToolRegistry", () => {
  it("should register a tool", () => {
    const registry = new ToolRegistry();
    const tool = new AddTool();
    registry.register(tool);
    expect(registry.names()).toContain("add");
  });

  it("should get a tool by name", () => {
    const registry = new ToolRegistry();
    const tool = new AddTool();
    registry.register(tool);
    const found = registry.get("add");
    expect(found).toBeDefined();
    expect(found?.name).toBe("add");
  });

  it("should return all tools", () => {
    const registry = new ToolRegistry();
    registry.register(new AddTool());
    const all = registry.all();
    expect(all.length).toBe(1);
  });
});

describe("Result", () => {
  it("should serialize to JSON", () => {
    const result = new Result(true, 42);
    const json = result.toJSON();
    expect(json).toHaveProperty("success", true);
    expect(json).toHaveProperty("data", 42);
  });
});